Source/WebCore/rendering/RenderLayerFilters.cpp

```diff
@@ -160,7 +160,7 @@ GraphicsContext* RenderLayerFilters::beginFilterEffect(RenderElement& renderer,
     if (!m_filter || m_targetBoundingBox != targetBoundingBox) {
         m_targetBoundingBox = targetBoundingBox;
         // FIXME: This rebuilds the entire effects chain even if the filter style didn't change.
-        m_filter = CSSFilter::create(renderer, renderer.style().filter(), m_renderingMode, m_filterScale, Filter::ClipOperation::Unite, m_targetBoundingBox, context);
+        m_filter = CSSFilter::create(renderer, renderer.style().filter(), m_filterMode, m_filterScale, Filter::ClipOperation::Unite, m_targetBoundingBox, context);
```

Comment:
The `mode` argument to `CSSFilter::create` should be a `preferredMode`, right? Somewhere there should be some code that loops over the list of filters and computes which mode can be used for that set of filters. Then that's the mode that actually gets used.

Reply:
This happened in `CSSFilter::create()` after building all the FilterFunctions by this statement:

```
    if (filterMode != FilterMode::Software && !filter->supportedFilterModes().contains(filterMode))
        filter->setFilterMode(FilterMode::Software);

```